Automattic/wp-calypso
# Plugins: Update how translations are handled in the action modal

This PR replaces the `getPluginActionDailog` method with a new hook that responds to changes in the user's locale via `useTranslate`. It also rewrites some plugin management-related code to reduce complexity and improve readability. All existing functionality should be preserved.

Resolves `1202619025189113-as-1202978931490015`.
Relates to #67799.

## Proposed Changes

* Create a new `useShowPluginActionDialog` hook (with supporting hook `useGetDialogText`).
* Create a higher-order component `withShowPluginActionDialog`, for a couple areas dependent on this functionality that don't (yet) support hooks.
* Add unit test coverage for both new hooks.
* Remove unused/unreferenced methods in the `PluginsList` component, and re-order/refactor some methods to reduce nesting, improve readability, and take advantage of new JavaScript features (e.g., `Array.flat`).
* Replace all references to the `getPluginActionDailog` method with usages of `useShowPluginActionDialog` or the HOC equivalent.
* Remove the old (and now, unused) `getPluginActionDailog` method.

## Testing Instructions

***TL;DR:** Regression testing for all plugin management functions.*

* In the Jetpack Cloud dashboard for agencies/partners, experiment with the various available plugin management actions:
  * Try affecting just one plugin, just one site, multiples of either, or multiples of both.
  * Try accepting the confirmation modal, and try canceling it.
  * Pay special attention to the modal's verbiage to check that the action name, site title and/or plugin name (if applicable), and spelling are all correct.
* In Calypso Blue, do much the same testing on the `/plugins/{pluginSlug}` and `/plugins/{pluginSlug}/{siteSlug}` pages. (NOTE: You'll likely need access to a WPCOM Business or eCommerce site for this step.) Confirm that all functionality works as expected.

### Reference screenshots

<img width="597" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/d3cdacb9-24a5-4e9d-b4b4-9faf7f21a1ca">

<img width="918" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/40484051-ee7a-4615-8db4-e5b04b1256cc">

<img width="1202" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/7d6f9da7-942e-4ad4-acf1-120579a43ee0">

<img width="1318" alt="image" src="https://github.com/Automattic/wp-calypso/assets/670067/69f9bf94-36fa-4fe3-a386-810267c6119f">

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202978931490015
